src/writeData/components/PluginAddToExistingConfiguration/Customize.tsx

```diff
@@ -0,0 +1,167 @@
+              {
+                name: pluginConfigName,
+                configured: ConfigurationState.Configured,
+                active: false,
+                plugin: {
+                  name: pluginConfigName,
+                  type: 'input',
+                },
+              },
+            ])
+          },
+          error => handleError(error)
+        )
+      } catch (error) {
+        handleError(error)
+      }
+    } else {
+      setIsValidConfiguration(true)
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
+  const handleChangeConfig = config => setPluginConfig(config)
+
+  const handleNameInput = (event: ChangeEvent<HTMLInputElement>) => {
+    onSetTelegrafConfigName(event.target.value)
+  }
+
+  const handleDescriptionInput = (event: ChangeEvent<HTMLInputElement>) => {
+    onSetTelegrafConfigDescription(event.target.value)
+  }
+
+  return (
+    <>
+      <Grid testID="plugin-edit-configuration-customize">
+        <Grid.Row className="plugin-edit-configuration-name-row">
+          <Form.Element label="Configuration Name">
+            <Input
+              type={InputType.Text}
+              value={telegrafConfigName}
+              name="name"
+              onChange={handleNameInput}
+              placeholder="Your configuration name"
+              titleText="Configuration Name"
+              size={ComponentSize.Medium}
+              autoFocus={true}
+              testID="plugin-edit-configuration-customize-input--name"
+            />
+          </Form.Element>
+        </Grid.Row>
+        <Grid.Row className="plugin-edit-configuration-description-row">
+          <Form.Element label="Configuration Description">
+            <Input
+              type={InputType.Text}
+              value={telegrafConfigDescription}
+              name="description"
+              onChange={handleDescriptionInput}
+              placeholder="Your configuration description"
+              titleText="Description (Optional)"
+              size={ComponentSize.Medium}
+              autoFocus={true}
+              testID="plugin-edit-configuration-customize-input--description"
+            />
+          </Form.Element>
+        </Grid.Row>
+        <Grid.Row className="plugin-edit-configuration--editor">
+          <div className="config-overlay">
+            <TelegrafConfig
+              config={pluginConfig}
+              onChangeConfig={handleChangeConfig}
+            />
+          </div>
+        </Grid.Row>
+      </Grid>
+    </>
+  )
+}
+
+const mstp = (state: AppState) => {
+  const {
+    telegrafConfigDescription,
+    telegrafConfigID,
+    telegrafConfigName,
+  } = getDataLoaders(state)
+  const telegrafs = getAll<Telegraf>(state, ResourceType.Telegrafs)
+
+  const selectedTelegraf = telegrafs?.find(
+    telegraf => telegraf.id === telegrafConfigID
+  )
+  return {
+    telegrafConfig: selectedTelegraf?.config ?? '',
+    telegrafConfigDescription,
+    telegrafConfigName,
+  }
```

Comment:
I think there might be some opportunity here to memoize these selectors as well

Reply:
i tried. Memoizing here has no effect. This component has the input fields for name and description, and also the editor for the configuration, all of which the user can change. `mstp` is responsible for updating the props to this component when the state changes. The state changes whenever the user changes any of the input fields or the contents of the editor.